File: DSA/VenikaPanwar_590013755/Day_59_Question_3.java
import java.util.*;

public class CheckIdenticalHeaps {
    public static boolean areHeapsIdentical(int[] heap1, int[] heap2) {
        if (heap1.length != heap2.length) return false;
        for (int i = 0; i < heap1.length; i++) {
            if (heap1[i] != heap2[i]) return false;
        }
        return true;
    }

    public static void main(String[] args) {
        int[] arr1 = {10, 9, 8};
        int[] arr2 = {10, 9, 8};
        int[] arr3 = {10, 8, 9};

        System.out.println(areHeapsIdentical(arr1, arr2)); // true
        System.out.println(areHeapsIdentical(arr1, arr3)); // false
    }
}

File: DSA/SAMAGRA_590014891/Day_59_Question_3.c
#include <stdio.h>
#include <stdbool.h>

bool areHeapsIdentical(int arr1[], int arr2[], int n1, int n2) {
    if (n1 != n2) return false;
    for (int i = 0; i < n1; i++) {
        if (arr1[i] != arr2[i]) return false;
    }
    return true;
}

int main() {
    int arr1[] = {10, 9, 8};
    int arr2[] = {10, 9, 8};
    int arr3[] = {10, 8, 9};

    printf("%s\n", areHeapsIdentical(arr1, arr2, 3, 3) ? "true" : "false");
    printf("%s\n", areHeapsIdentical(arr1, arr3, 3, 3) ? "true" : "false");
    return 0;
}
